### PR TITLE
style: use f-strings instead of %s-formatting

### LINKED ISSUE #1240: replace `'%s'` with new string formatting f`{var}` with best reasonable effort
**Describe your problem**

##Good first issue##

In the current codebase, there are two types of string formating, would be nice to replace `'%s'` with new string formatting f`{var}` with the best reasonable effort.

**What is your guess?**

---

**Environment**

**Screenshots**

### PR DESCRIPTION
**Pull request by cristianmtr**
Closes #1240 

- tested by running various examples
- used PyCharm automatic replace f-string. This worked in most cases. Some required manual rewriting

### CODE REVIEW

**Review by hanxiao — COMMENTED** [review]
nicely done! 👍

**Comment by hanxiao** [comment]
i reviewed the first 9 files but distracted by something else, seems ok so far. would be nice if someone can take over. otherwise, we do a lazy consensus in a few hours.

File: jina/clients/python/grpc.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ def __init__(self, args: 'argparse.Namespace'):
         self.logger.debug('setting up grpc insecure channel...')
         # A gRPC channel provides a connection to a remote gRPC server.
         self._channel = grpc.insecure_channel(
-            '%s:%d' % (args.host, args.port_expose),
+            f'{args.host}:{args.port_expose:d}',
```
Comment (JoanFM) [inline jina/clients/python/grpc.py:33]: this `:d` seems odd

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jina/checker.py [hunk 2]
```diff
@@ -55,12 +55,12 @@ def __init__(self, args: 'argparse.Namespace'):
         from jina.logging.profile import TimeContext
         from google.protobuf.json_format import MessageToJson
         import time
-        ctrl_addr = 'tcp://%s:%d' % (args.host, args.port)
+        ctrl_addr = f'tcp://{args.host}:{args.port:d}'
```
Comment (JoanFM) [inline jina/checker.py:58]: same here with the `:d`

File: jina/checker.py [hunk 3]
```diff
@@ -55,12 +55,12 @@ def __init__(self, args: 'argparse.Namespace'):
...
         try:
             total_time = 0
             total_success = 0
             for j in range(args.retries):
-                with TimeContext('ping %s at %d round' % (ctrl_addr, j), default_logger) as tc:
+                with TimeContext(f'ping {ctrl_addr} at {j:d} round', default_logger) as tc:
```
Comment (JoanFM) [inline jina/checker.py:63]: and here

File: jina/peapods/pea.py [hunk 4]
```diff
@@ -139,7 +139,7 @@ def __init__(self, args: Union['argparse.Namespace', Dict]):
             if self.args.name:
                 self.name = self.args.name
             if self.args.role == PeaRoleType.PARALLEL:
-                self.name = '%s-%d' % (self.name, self.args.pea_id)
+                self.name = f'{self.name}-{self.args.pea_id:d}'
```
Comment (JoanFM) [inline jina/peapods/pea.py:142]: here

**Review by JoanFM — COMMENTED** [review]
I am not sure if it is a problem or a `formatting` question

File: tests/unit/test_yamlparser.py [hunk 5]
```diff
@@ -73,7 +73,7 @@ class DummyClass:
     assert type(a) == DummyClass
 
     with open(resource_filename('jina',
-                                '/'.join(('resources', 'executors.requests.%s.yml' % 'BaseExecutor')))) as fp:
+                                '/'.join(('resources', f'executors.requests.{"BaseExecutor"}.yml')))) as fp:
```
Comment (JoanFM) [inline tests/unit/test_yamlparser.py:76]: I think here u can add directly `'executors.requests.BaseExecutor.yml'` right?

File: jina/peapods/zmq.py [hunk 6]
```diff
@@ -88,7 +88,7 @@ def get_ctrl_address(args: 'argparse.Namespace') -> Tuple[str, bool]:
         if ctrl_with_ipc:
             return _get_random_ipc(), ctrl_with_ipc
         else:
-            return 'tcp://%s:%d' % (args.host, args.port_ctrl), ctrl_with_ipc
+            return f'tcp://{args.host}:{args.port_ctrl:d}', ctrl_with_ipc
```
Comment (JoanFM) [inline jina/peapods/zmq.py:91]: and here

File: jina/parser.py [hunk 7]
```diff
@@ -714,10 +714,10 @@ def _metavar_formatter(self, action, default_metavar):
 
             if len(action.choices) > 4:
                 choice_strs = ', '.join([str(c) for c in action.choices][:4])
-                result = '{%s ... %d more choices}' % (choice_strs, len(action.choices) - 4)
+                result = f'{{{choice_strs} ... {len(action.choices) - 4:d} more choices}}'
```
Comment (JoanFM) [inline jina/parser.py:717]: also here?

File: jina/clients/python/grpc.py (hunk 1 above)
Comment (cristianmtr) [reply jina/clients/python/grpc.py:33]: It's a formatting code, like the 's' and 'f'. But yeah, it doesn't add anything. I can remove them

File: tests/unit/test_yamlparser.py (hunk 5 above)
Comment (cristianmtr) [reply tests/unit/test_yamlparser.py]: Haha, strange that it was there

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Review by deepankarm — APPROVED** [review]
Had no idea editors can do this. Really nice.

**Review by JoanFM — APPROVED** [review]

**Pull request merged**